sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/ExpansionService.java

```diff
@@ -391,17 +410,19 @@ default InputT createInput(Pipeline p, Map<String, PCollection<?>> inputs) {
         request.getTransform().getSpec().getUrn());
     LOG.debug("Full transform: {}", request.getTransform());
     Set<String> existingTransformIds = request.getComponents().getTransformsMap().keySet();
-    Pipeline pipeline = Pipeline.create();
-    ExperimentalOptions.addExperiment(
-        pipeline.getOptions().as(ExperimentalOptions.class), "beam_fn_api");
-    // TODO(BEAM-10670): Remove this when we address performance issue.
-    ExperimentalOptions.addExperiment(
-        pipeline.getOptions().as(ExperimentalOptions.class), "use_sdf_read");
+    Pipeline pipeline = createPipeline();
+    if (!ExperimentalOptions.hasExperiment(pipelineOptions, "use_deprecated_read")) {
```

Comment:
Where are these options read? Do we test that these options are respected?

Reply:
Not as part of this PR, but I believe there are tests for that. We need it so that we don't override `use_deprecated_read` with `use_sdf_read` (opt-out, then opt-in).

Reply:
:+1: